pyincore_viz/geoutil.py

```diff
@@ -1079,3 +1082,141 @@ def get_ipyleaflet_heatmap(locations=None, radius=10, blur=10, max=1, name=""):
         heatmap.gradient = {0.4: 'red', 0.6: 'yellow', 0.7: 'lime', 0.8: 'cyan', 1.0: 'blue'}
 
         return heatmap
+
+    @staticmethod
+    def plot_choropleth_multiple_fields_from_single_dataset(dataset, field_list, zoom_level=10):
+        in_gpd = None
+        center_x = None
+        center_y = None
+
+        # check if the dataset is geodataset and convert dataset to geodataframe
+        try:
+            in_gpd = gpd.read_file(dataset.local_file_path)
+            center_x = in_gpd.bounds.minx.mean()
+            center_y = in_gpd.bounds.miny.mean()
+        except Exception:
+            raise("Given dataset is not a geodataset")
+
+        # skim geodataframe only for needed fields
+        field_list.append('geometry')
+        in_gpd_tmp = in_gpd[field_list]
+        geo_data_dic = json.loads(in_gpd_tmp.to_json())
+
+        out_map = ipylft.Map(center=(center_y, center_x), zoom=zoom_level,
+                             crs=projections.EPSG3857, scroll_wheel_zoom=True)
+
+        for fld in field_list:
+            if fld != 'geometry':
+                tmp_choro_data = GeoUtil.create_choro_data_from_pd(in_gpd, fld)
+                # add choropleth data to  map
+                tmp_layer = ipylft.Choropleth(
+                    geo_data=geo_data_dic,
+                    choro_data=tmp_choro_data,
+                    colormap=linear.YlOrRd_04,
+                    border_color='black',
+                    style={'fillOpacity': 0.8},
+                    name=fld
+                )
+
+                out_map.add_layer(tmp_layer)
+
+        out_map.add_control(ipylft.LayersControl(position='topright'))
+        out_map.add_control(ipylft.FullScreenControl(position='topright'))
+
+        return out_map
+
+    @staticmethod
+    def plot_choropleth_multiple_dataset(dataset_list, field_list, zoom_level=10):
+        geodata_dic_list = []
+        choro_data_list = []
+        bbox = None
+
+        # check the size of dataset list and field list
+        if len(dataset_list) != len(dataset_list):
+            raise Exception("The dataset list size and field list size doesn't match")
+
+        # check if the dataset is geodataset and convert dataset to geodataframe
+        for dataset, fld in zip(dataset_list, field_list):
+            try:
+                tmp_gpd = gpd.read_file(dataset.local_file_path)
+                tmp_fld_list = []
+                tmp_min_x = tmp_gpd.bounds.minx.mean()
+                tmp_min_y = tmp_gpd.bounds.miny.mean()
+                tmp_max_x = tmp_gpd.bounds.maxx.mean()
+                tmp_max_y = tmp_gpd.bounds.maxy.mean()
+
+                if bbox is None:
+                    bbox = [tmp_min_x, tmp_min_y, tmp_max_x, tmp_max_y]
```

Comment:
isn't bbox always None cuz you define that in line 1132?

Reply:
it will be None at the very beginning of the iteration but it will not after the first iteration.

Reply:
it belongs to iteration so it will be changed.

Reply:
make sense. Thank you!